kcidb/monitor/spool/__init__.py

```diff
@@ -97,29 +96,46 @@ def post(self, notification, timestamp=None):
             timestamp:      An "aware" datetime.datetime object specifying the
                             notification creation time, or None to use
                             datetime.datetime.now(datetime.timezone.utc).
+            due:            An "aware" datetime.datetime object specifying the
+                            time notification should be sent out,
+                            or None to use "timestamp".
 
         Returns:
-            True, if the notification was posted onto the spool,
-            False, if not (it was already there).
+            True, if the new notification was posted onto the spool,
+            or the existing, "unpicked" notification was updated,
+            False, if not (it existed and was "picked" at the moment).
         """
         assert isinstance(notification, Notification)
         assert timestamp is None or \
             isinstance(timestamp, datetime.datetime) and timestamp.tzinfo
         if timestamp is None:
             timestamp = datetime.datetime.now(datetime.timezone.utc)
+        message = notification.render().as_string(policy=email.policy.SMTPUTF8)
+        due = notification.message.due
+        doc = self._get_doc(notification.id)
 
-        try:
-            self._get_doc(notification.id).create(dict(
+        @firestore.transactional
+        def create_or_update(transaction, message, doc, due):
+            snapshot = doc.get(field_paths=["picked_until"],
+                               transaction=transaction)
+            if snapshot.exists:
+                # if the notification has been picked
+                if snapshot.get("picked_until") > timestamp:
+                    return False
+                transaction.update(doc, dict(message=message, due=due))
+                return True
```

Comment:
Instead of having `return True` here, could you instead use two branches of the preceding `if` statement, so that we have something like this:
```
if exists:
    update
else:
    create
```
This way we will only have one successful exit point in the function, and if we need to add something to the successful exit path we have only one place to update, and less of a chance of forgetting to add it to this one as well?